solidstart-tanstackquery-tailwind-modules/src/utils/onclickOutside.ts

```diff
@@ -0,0 +1,10 @@
+import { onCleanup } from 'solid-js';
+
+export const clickOutside = (el: any, accessor: any) => {
+  const onClick = (e: any) => {
+    !el.contains(e.target) && accessor()?.();
+  };
+  document.body.addEventListener('click', onClick);
+
+  onCleanup(() => document.body.removeEventListener('click', onClick));
+};
```

Comment:
At least we can typed the el and e, never see accessor, don't know what type it is :-D

```suggestion
export const clickOutside = (el: HTMLDivElement, accessor: any) => {
  const onClick = (e: MouseEvent) => {
    !el.contains(e.target as unknown as HTMLDivElement) && accessor()?.();
  };
  document.body.addEventListener('click', onClick);

  onCleanup(() => document.body.removeEventListener('click', onClick));
};
```